logo.js

```diff
@@ -3693,4 +3693,47 @@ function LogoInterpreter(turtle, stream, savehook)
       throw new Error("Internal error: Unbound procedure");
     return value;
   });
+
+  //----------------------------------------------------------------------
+  //
+  // 9. Sound / Basic Music / Web Speech
+  //
+  //----------------------------------------------------------------------
+
+  var soundSeq = [];
+  var audioCtx = new (window.AudioContext || window.webkitAudioContext)();
+
+  function playSound(frequency, duration) {
+      const oscillator = audioCtx.createOscillator();
+
+      oscillator.type = 'square';
+      oscillator.frequency.value = frequency; // value in hertz
+      oscillator.connect(audioCtx.destination);
+      oscillator.start();
+
+      setTimeout(
+          function() {
+                  oscillator.stop();
+                  playSequence();
+          }, duration);
+  }
+
+  function playSoundSeq() {
+      if (soundSeq.length > 0) {
```

Comment:
It looks like you're intending that each call to `beep` gets queued, and the sounds play one at a time. But as written, there's nothing that stops a note from playing if a note is already playing. 

Consider leaving the note in the queue until it finishes playing, then popping and checking the queue again.

Reply:
I renamed the functions with ’queue‘ in api, and in fact I want to implements music play with some different tones. So far, only 1 thread was supported which is not enough... 

I commit this 1-thread version first.